### PR TITLE
Component/chart

### PR DESCRIPTION
**Pull request by mmfallacy**
This PR introduces the Chart component which serves as the primary data visualization for the flow rate.

### CODE REVIEW

File: src/components/Chart/Display.svelte [hunk 1]
```diff
@@ -0,0 +1,120 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+    import { flow } from './mock.ts';
+    import Chart from 'chart.js/auto';
+    import type { ChartConfiguration } from 'chart.js';
+    import 'chartjs-adapter-date-fns';
+    import ChartStreaming from 'chartjs-plugin-streaming';
+    import ChartZoom from 'chartjs-plugin-zoom';
+    import colors from 'tailwindcss/colors';
```
Comment (BastiDood) [inline src/components/Chart/Display.svelte:9]: Just reordering some imports here. I basically alphabetized by the name of the third-party package.
```suggestion
    import type { ChartConfiguration } from 'chart.js';
    import Chart from 'chart.js/auto';
    import ChartStreaming from 'chartjs-plugin-streaming';
    import ChartZoom from 'chartjs-plugin-zoom';
    import { onMount } from 'svelte';
    import colors from 'tailwindcss/colors';

    import 'chartjs-adapter-date-fns';

    import { flow } from './mock.ts';
```

File: src/routes/Dashboard/Page.svelte
Comment (BastiDood) [inline src/routes/Dashboard/Page.svelte:1]: There is apparently one usage of `Spacer` that is declared but never used. The indentation is also inconsistent in the imports.

File: src/components/Chart/Display.svelte [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        },
+    };
+
+    $: {
+        // Update chart upon retrieving data and clear buffer.
+        const chart = Chart.getChart(canvas);
+        chart?.data.datasets[0]?.data.push(
+            ...$flow.map(flow => ({ x: flow.ts.getTime(), y: flow.flow }))
+        );
+        $flow = [];
```
Comment (BastiDood) [inline src/components/Chart/Display.svelte:110]: I believe we may use mutation here to make it more efficient.
```suggestion
        const points = $flow
            .splice(0)
            .map(({ ts, flow }) => ({
                x: ts.getTime(),
                y: flow,
            }));
        $flow = $flow; // notify mutation
        chart?.data.datasets[0]?.data.push(...points);
```

**Review by BastiDood — CHANGES_REQUESTED** [review]

File: src/routes/Dashboard/Page.svelte
Comment (mmfallacy) [reply src/routes/Dashboard/Page.svelte:1]: Removed in [cd3601b](https://github.com/drippy-iot/dashboard/pull/15/commits/cd3601b671016fc6fb33c8c2b853e4183bfb4606)

**Review by BastiDood — APPROVED** [review]

**Pull request merged**